dask/bag/core.py

```diff
@@ -1134,7 +1134,7 @@ def partition(grouper, sequence, npartitions, p, nelements=2**20):
         d2 = defaultdict(list)
         for k, v in d.items():
             d2[abs(hash(k)) % npartitions].extend(v)
-        p.append(d2)
+        p.append(d2, fsync=True)
```

Comment:
I've been running into similar issues lately that seem to persist even with this fix.  Tracking progress here: https://github.com/dask/dask/pull/1383#issuecomment-234721455